node/src/integration-test/kotlin/net/corda/node/utilities/registration/NodeRegistrationTest.kt

```
@@ -57,18 +65,27 @@ class NodeRegistrationTest {
         server.close()
     }
 
-    // TODO Ideally this test should be checking that two nodes that register are able to transact with each other. However
-    // starting a second node hangs so that needs to be fixed.
     @Test
     fun `node registration correct root cert`() {
         val compatibilityZone = CompatibilityZoneParams(URL("http://$serverHostAndPort"), rootCert = rootCertAndKeyPair.certificate.cert)
         internalDriver(
                 portAllocation = portAllocation,
-                notarySpecs = emptyList(),
-                compatibilityZone = compatibilityZone
+                compatibilityZone = compatibilityZone,
+                extraCordappPackagesToScan = listOf("net.corda.finance")
         ) {
-            startNode(providedName = CordaX500Name("Alice", "London", "GB")).getOrThrow()
-            assertThat(registrationHandler.idsPolled).contains("Alice")
+            val ALICE_NAME = "Alice"
+            val GENEVIEVE_NAME = "Genevieve"
+            val alice = startNode(providedName = CordaX500Name(ALICE_NAME, "London", "GB")).getOrThrow()
+            assertThat(registrationHandler.idsPolled).contains(ALICE_NAME)
+            val genevieve = startNode(providedName = CordaX500Name(GENEVIEVE_NAME, "London", "GB")).getOrThrow()
+            assertThat(registrationHandler.idsPolled).contains(GENEVIEVE_NAME)
+
+            // Check we nodes communicate among themselves (and the notary).
+            val anonymous = true
+            genevieve.rpc.startFlow(::CashIssueAndPaymentFlow, 1000.DOLLARS, OpaqueBytes.of(12),
+                    alice.nodeInfo.legalIdentities.first(),
+                    anonymous,
+                    defaultNotaryIdentity)
```

`startFlow` returns a Future which must be waited on otherwise any exception is lost